nbval/plugin.py

```diff
@@ -243,8 +256,22 @@ def collect(self):
                 # The cell may contain a comment indicating that its output
                 # should be checked or ignored. If it doesn't, use the default
                 # behaviour. The --nbval option checks unmarked cells.
-                options = defaultdict(bool, find_metadata_tags(cell.metadata))
-                options.update(find_comment_markers(cell.source))
+                with warnings.catch_warnings(record=True) as ws:
+                    options = defaultdict(bool, find_metadata_tags(cell.metadata))
+                    comment_opts = dict(find_comment_markers(cell.source))
+                    if set(comment_opts.keys()) & set(options.keys()):
+                        warnings.warn(
+                            "Overlapping options from comments and metadata, "
+                            "using options from comments: %s" %
+                            str(set(comment_opts.keys()) & set(options.keys())))
+                    for w in ws:
+                        self.parent.config.warn(
+                            "C1",
+                            w.message,
+                            '%s:Cell %d' % (
+                                getattr(self, "fspath", None),
+                                cell_num))
```

Comment:
I added a version specifier in setup.py, which makes it go and update py.test to 3.0.6, which gives a much more interesting error. Any input on what it can be?